assets/js/src/utils/index.js

```diff
@@ -26,6 +26,23 @@ export const getProductFieldObject = ( product, quantity ) => {
 		),
 		...variantData,
 	};
+
+	// Apply discounts to ecommerce events.
+	// https://developers.google.com/analytics/devguides/collection/ga4/apply-discount?client_type=gtag
+	if (
+		typeof product.price_after_coupon_discount === 'number' &&
+		! isNaN( product.price_after_coupon_discount ) &&
+		product.price_after_coupon_discount !== product.prices.price
```

Comment:
WDYT about changing this to `product.price_after_coupon_discount < product.prices.price` I don't think we can have negative discounts.